### PR TITLE
Optimize document JSON structure for AI validation efficiency

### PR DESCRIPTION
**Pull request by andtankian**
### 🧾 Summary

Adds optimized document JSON structure to reduce payload size sent to AI validation service by transforming full Document objects into streamlined OptimizedDocumentJson format with only essential fields.

### 🧩 Context

**Why this change?** The current AI validation service receives full Document objects containing extensive nested data, resulting in larger payloads than necessary for validation tasks.

**Problem it solves:** Large document payloads increase network overhead, processing time, and costs when sent to external AI validation services.

**Business value:** Improves AI validation performance and reduces operational costs while maintaining validation accuracy.

### 🧱 Scope of this PR

**This PR includes:**

- [x] New features
- [ ] Bug fixes
- [ ] Refactoring
- [ ] Documentation updates
- [ ] Configuration changes

**New features added:**
- OptimizedDocumentJson interface with streamlined document structure
- OptimizedAddress, OptimizedParticipant, and OptimizedDocumentEvent interfaces
- optimizeDocumentJsonForValidation helper function
- Comprehensive test suite for document optimization logic
- Integration of optimization in AI validation service

**Intentionally excluded:** Performance metrics collection (will be added in follow-up PR)

### 🧪 How to test

1. Run the new test suite: `pnpm nx test shared-methodologies-ai-attachment-validator`
2. Verify optimization logic with existing documents
3. Test AI validation still works with optimized payloads
4. Check that all required document fields are preserved in optimized format

### 🧠 Considerations for Review

**Key files to review:**

- `libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.api.dto.ts` - New optimized interfaces
- `libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts` - Core optimization logic
- `libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.spec.ts` - Comprehensive test coverage
- `libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.service.ts` - Service integration

**Focus areas:**
- Verify all essential document data is preserved in optimization
- Review test coverage for edge cases (empty events, missing optional fields)
- Confirm type safety with OptimizedDocumentJson interfaces

---

### ✅ Pre-merge Checklist

- [x] Code follows project style guidelines and naming conventions
- [x] All new code has appropriate test coverage  
- [x] Linter warnings and errors have been resolved
- [x] No debugging code or console logs remain
- [x] TypeScript strict mode compliance verified
- [x] Unit tests updated and passing
- [ ] End-to-end tests updated — N/A (no user-facing changes)
- [x] Documentation updated — N/A (internal optimization)

## Summary by CodeRabbit

- **New Features**
  - Added document pre-processing that produces compact "optimized" document shapes (addresses, participants, events, values) and updated the validator to send these optimized payloads; also added a new exported assertion for optimized documents.
- **Bug Fixes**
  - Validation error formatting improved: empty/null reasons are omitted so fields no longer show trailing ": ".
- **Tests**
  - Added comprehensive tests for minimal/complete documents, events (with/without attachments), nullable fields, and mapping of addresses/participants.

### CODE REVIEW

File: libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts [hunk 1]
```diff
@@ -1,6 +1,152 @@
...
+  addresses[primaryAddress.id] = {
+    city: primaryAddress.city,
+    country: primaryAddress.countryCode,
+    lat: primaryAddress.latitude,
+    lng: primaryAddress.longitude,
+    num: primaryAddress.number,
+    state: primaryAddress.countryState,
+    street: primaryAddress.street,
+    zip: primaryAddress.zipCode,
+  };
```
[the remark this answers is not in the document]
Comment (andtankian) [reply libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts:30]: Actually, for the current flow where this piece will be called, the address is always present. The create type also reflects that (it's not optional)  What do you think?

[the remark this answers is not in the document]
Comment (andtankian) [reply libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts:30]: Regarding type casting. We avoid casting as much as possible.  That being said, if you're ok with the answer you may resolve this review comment.

File: libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts [hunk 2]
```diff
@@ -1,6 +1,152 @@
...
+    }
+  }
+
+  return participants;
+}
+
 export const formatInvalidField = (
   fieldName: NonEmptyString,
   reason: NonEmptyString | null,
 ): string => `${fieldName}: ${reason}`;
```
[the remark this answers is not in the document]
Comment (andtankian) [reply libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts]: @coderabbitai, can you take a look if this commit fixes the issue? 8558905

File: libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.api.dto.ts [hunk 3]
```diff
@@ -19,6 +24,61 @@ export interface ApiValidateAttachmentResponse {
...
+  externalId?: NonEmptyString | undefined;
+  id: NonEmptyString;
+  label?: NonEmptyString | undefined;
+  meta?:
+    | undefined
+    | {
+        [key: string]: unknown;
+      };
+  name: NonEmptyString;
+  pId: NonEmptyString;
```
Comment (cris-santos) [inline libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.api.dto.ts]: What's `aId` and `pId`? If it's `addressId` and `participantId`, unless it have a significant impact on the cost, I would keep it more descriptive. If we keep as `aId` and `pId`, it may be helpful to add a comment with the original field name

File: libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts [hunk 4]
```diff
@@ -1,6 +1,153 @@
...
+      participants[participant.id as NonEmptyString] = {
+        country: participant.countryCode as NonEmptyString,
+        name: participant.name as NonEmptyString,
+        taxId: participant.piiSnapshotId as NonEmptyString,
+        type: participant.type as NonEmptyString,
+      };
+    }
+  }
+
+  return participants;
```
Comment (cris-santos) [inline libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts]: Just to make sure, is it safe and required to make these type castings? Shouldn't we validate the input/output with typia to ensure the types?

File: libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.api.dto.ts (hunk 3 above)
Comment (andtankian) [reply libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.api.dto.ts]: Yes! We're explicitly setting these fields to a shorter version of it due cost impact. I don't like these non-descriptive fields as well but we decreased the cost by 70% ddoing these optimations. So I'll keep it. 
It's a great idea to comment the fields with original field name. 28d3644

File: libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts [hunk 5]
```diff
@@ -1,6 +1,153 @@
...
+    id: event.id,
+    ...(event.label && { label: event.label }),
+    ...(event.metadata && {
+      meta: event.metadata as { [key: string]: unknown },
+    }),
+    name: event.name,
+    pId: event.participant.id,
+    ...(event.value !== undefined && { value: event.value }),
+  };
+}
```
[the remark this answers is not in the document]
Comment (andtankian) [reply libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts]: Can you look at: 791e27c

File: libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts [hunk 6]
```diff
@@ -1,6 +1,153 @@
...
+  events?: DocumentEvent[],
+): Record<NonEmptyString, OptimizedParticipant> {
+  const participants: Record<NonEmptyString, OptimizedParticipant> = {};
+
+  participants[primaryParticipant.id as NonEmptyString] = {
+    country: primaryParticipant.countryCode as NonEmptyString,
+    name: primaryParticipant.name as NonEmptyString,
+    taxId: primaryParticipant.piiSnapshotId as NonEmptyString,
+    type: primaryParticipant.type as NonEmptyString,
+  };
```
[the remark this answers is not in the document]
Comment (andtankian) [reply libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts]: I removed the casting and used typia to validate the fields. It's team convention. Can you check?

File: libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.spec.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+      const event = result.events?.[0];
+
+      expect(event).toBeDefined();
+      expect(event?.id).toBeDefined();
+      expect(event?.name).toBeDefined();
+      expect(event?.aId).toBeDefined();
+      expect(event?.pId).toBeDefined();
+    }
+  });
```
[the remark this answers is not in the document]
Comment (andtankian) [reply libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.spec.ts]: a220ee4

[the remark this answers is not in the document]
Comment (andtankian) [reply libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.spec.ts]: Please, look again.

File: libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts (hunk 4 above)
Comment (andtankian) [reply libs/shared/methodologies/ai-attachment-validator/src/ai-attachment-validator.helpers.ts]: I've improved the types and added a typia assertion. Can you review it again?

**Review by cris-santos — APPROVED** [review]
